### ISSUE TITLE
Throwing errors in Function Nodes generates an overlapping null reference error

### ISSUE THREAD

**Issue by LordFokas** [node-red/node-red#2269]
### What are the steps to reproduce?
Create a Function Node;
`throw "This custom error that I want to catch elsewhere";`

### What happens?
The Function Node attempts to remove part of the call stack from the error, however due to running inside some kind of eval / sandbox / whatever, the thrown error has no call stack, which causes a very cryptic `TypeError: Cannot read property 'search' of undefined` here: https://github.com/node-red/node-red/blob/master/packages/node_modules/%40node-red/nodes/core/core/80-function.js#L241

This null / undefined reference error overlaps and supersedes the error I'm trying to catch and is ultimately what my Catch Node receives.

### What do you expect to happen?
I expect errors thrown in any kind of scripting nodes to be handed directly to the catch nodes.

### Please tell us about your environment:

- [x] Node-RED version: 0.20.7
- [x] Node.js version: 12.7
- [ ] npm version:
- [x] Platform/OS: RPi 3B+ running the latest Raspbian
- [ ] Browser:

**Comment by knolleary** [comment]
You say you are using node.js 0.12. so you really mean that? Or are you on 12.x ?

We do not support node 0.12 as it has been out of support for a long time.

If you are actually on node 12, then I suspect this is because you are throwing a string and not an error.

Try changing it to:

```
throw new Error("This custom error that I want to catch elsewhere")
```

**Comment by LordFokas** [comment]
Sorry, fixed it, of course it's **12.7**
And yes, I tend to throw strings, mostly because I can and it's simple.

Even though it's trivial for me to adopt this, since JS lets you throw anything you want to and not just actual errors, maybe Node-RED should support it anyways?

**Comment by dceejay** [comment]
Thanks - yes we could handle it better - fix pushed to dev.
